arch/xtensa/core/userspace.S

```diff
@@ -20,6 +22,37 @@
 .global	xtensa_do_syscall
 .align	4
 xtensa_do_syscall:
+#if XCHAL_HAVE_THREADPTR == 0
```

Comment:
Dumb question: but which hardware has an MPU but no THREADPTR?  This seems a little academic?   If it's just the qemu device, maybe we can reconfigure it?